accounts/abi/type.go

```diff
@@ -219,7 +219,12 @@ func NewType(t string, internalType string, components []ArgumentMarshaling) (ty
 		typ.T = FunctionTy
 		typ.Size = 24
 	default:
-		return Type{}, fmt.Errorf("unsupported arg type: %s", t)
+		if strings.Contains(internalType, "contract") {
```

Comment:
```suggestion
		if strings.HasPrefix(internalType, "contract ") {
```
doesn't hurt to make it a bit more strict, imo